src/TableFunctions/TableFunctionValues.cpp

```diff
@@ -83,7 +82,7 @@ void TableFunctionValues::parseArguments(const ASTPtr & ast_function, ContextPtr
     ASTs & args_func = ast_function->children;
 
     if (args_func.size() != 1)
-        throw Exception("Table function '" + getName() + "' must have arguments", ErrorCodes::LOGICAL_ERROR);
+        throw Exception("Table function '" + getName() + "' must have arguments", ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
```

Comment:
First of all, why not to replace it with formatted message like this?
```suggestion
        throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH, "Table function '{}' must have arguments", getName());
```